src/backend/document.hpp

```diff
@@ -22,12 +22,14 @@
 #include <giomm/file.h>
 #include <giomm/liststore.h>
 #include <poppler/cpp/poppler-document.h>
+#include <vector>
 
 namespace Slicer {
 
 class Document {
 public:
     Document(const Glib::RefPtr<Gio::File>& sourceFile);
+    Document(const std::vector<Glib::RefPtr<Gio::File>> sourceFiles);
```

Comment:
The `sourceFiles` parameter should be passed by reference, like in the other constructor above